src/common/types/timestamp.cpp

```diff
@@ -298,6 +298,9 @@ timestamp_t Timestamp::GetCurrentTimestamp() {
 
 timestamp_t Timestamp::FromEpochSeconds(int64_t sec) {
 	int64_t result;
+	if (!Timestamp::IsFinite(timestamp_t(sec))) {
```

Comment:
This won't work because `±infinity` are not an exact number of seconds.

I don't think these `IsFinite` tests should be here - they expose the internal representation of `infinity`. The test should be in the cast code, not here

Reply:
I'm confused, they are just the min and max points of the `int64_t` range right?
If the value came from lets say US, got converted to S and is now cast to NS, if it was infinite we want it to stay infinite right?

I'm just not sure why we don't want to handle this here, that covers everything.
We could add an assertion here instead, but then we'll likely have bug reports in the future for all the places that were missed or added later

Reply:
I see what you mean - infinite values are the same for all types, so I suppose it is OK. 

But there are places where we have to check beforehand anyway (e.g., data part extraction) so this will result in the test being applied twice. Maybe that is OK but it shows how different callers may have different needs.

Another set of examples are in the various adapter libraries (ODBC, Python, etc.) where the caller has to check for infinity anyway to map it to the appropriate value for the library (e.g., `9999-12-31 59:59:59.999999` for Python) So again the call will be made twice.

Reply:
So where did we land on this?
Are we fine with checking this at the conversion level just to make sure it's not missed?

Or do we want to try to be more performant by not (possibly) redundantly checking this, only checking it at the root?
I wonder if the performance suffers from this, it's only a shallow comparison of integers, but at the same time it's done for every value.

Reply:
@hawkfish 
So where did we land on this?
Are we fine with checking this at the conversion level just to make sure it's not missed?

Or do we want to try to be more performant by not (possibly) redundantly checking this, only checking it at the root?
I wonder if the performance suffers from this, it's only a shallow comparison of integers, but at the same time it's done for every value.